server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -171,6 +171,26 @@ export class AgenticChatController implements ChatHandlers {
             const toolUseId = params.messageId
             try {
                 await this.#undoFileChange(toolUseId)
+                const cachedToolUse = this.#triggerContext.getToolUseLookup().get(toolUseId)
+                if (cachedToolUse) {
+                    this.#features.chat.sendChatUpdate({
```

Comment:
What exactly does sendChatUpdate do? Is it force pushing this chat message to the client? Is there a reason we can't do this through the chatResultStream interface?

Reply:
My understanding is this is used for updating an existing message with messageId. I guess since this isn't a progress message we shouldn't use chatResultStream? Not sure why one is under `this.#features.chat` and the other is under `this.#features.lsp` though